src/App.tsx

```diff
@@ -15,48 +15,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET: {
+        return a.localeCompare(b);
+      }
+
+      case SortType.LENGTH: {
+        return a.length - b.length;
+      }
+
+      case SortType.NONE: {
+        return 0;
+      }
+
+      default: {
+        throw new Error('Wrong sorting type!');
+      }
+    }
+  });
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortedByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const wasChanged = isReversed
+    || sortType === SortType.ALPHABET
+    || sortType === SortType.LENGTH;
```

Comment:
```suggestion
  const wasOrderChanged = sortType !== SortType.NONE || isReversed
```
you can simplify it like this